### PR TITLE
Improve assoc array docs

### PR DESCRIPTION
**Pull request by dereuromark**
and update phpstan to lastest version.

Only works after https://github.com/cakephp/cakephp/runs/3988894578?check_suite_focus=true#step:10:33 is merged as the memory leak would then not appear anymore.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -110,7 +110,7 @@
             "@psalm"
         ],
         "stan-tests": "phpstan.phar analyze -c tests/phpstan.neon",
-        "stan-setup": "cp composer.json composer.backup && composer require --dev symfony/polyfill-php81 phpstan/phpstan:0.12.96 psalm/phar:~4.11.0 && mv composer.backup composer.json",
+        "stan-setup": "cp composer.json composer.backup && composer require --dev symfony/polyfill-php81 phpstan/phpstan:0.12.99 psalm/phar:~4.11.0 && mv composer.backup composer.json",
```
Comment (ADmad) [inline composer.json:113]: This release still has memory leak and will crash the container. Need to wait for 1.0 for the issue to be fixed.

Comment (dereuromark) [reply composer.json:113]: I already fixed it as per the authors suggestion in the follow up PR https://github.com/cakephp/cakephp/pull/16069

Comment (dereuromark) [reply composer.json:113]: Unfortunately, also tools like rector use phpstan internally, so it also blocks other tools from being used as of right now.

**Comment by dereuromark** [comment]
Only  2152 iterable issues more to go :P

**Pull request merged**